enhancements/service-catalog/service-catalog-and-broker-removal.md

```diff
@@ -87,69 +84,103 @@ cluster-svcat-controller-manager-operator, with a [Job](https://kubernetes.io/do
 This might be accomplished with just one of the operators, that would need
 a little more investigation.
 
-Replace the operator's code with a new Job executable that will perform the
-deletion of the apiserver associated with the Service Catalog. The following items
-would be removed:
-
-- ClusterServiceClasses
-- ClusterServicePlans
-- ClusterServiceBrokers
-- ServiceClasses
-- ServicePlans
-- ServiceBrokers
-- ServiceInstances
-- ServiceBindings
-
 The idea is that the Job will run during the upgrade until completion. Once
-finished, the Service Catalog will have been removed from the cluster.
-
-The Service Catalog operators would *remain* in the release payload but the
+finished, the Service Catalog Operators will have been removed from the
+cluster. The Job that will delete the Service Catalog operators and their
+associated items.
+
+- cluster-svcat-apiserver-operator
+  - Role named `prometheus-k8s in `openshift-service-catalog-apiserver-operator`
+  - RoleBinding named `prometheus-k8s in `openshift-service-catalog-apiserver-operator`
+  - ServiceMonitor in `openshift-service-catalog-apiserver-operator`
+  - Namespace `openshift-service-catalog-apiserver-operator` (not sure
+    if we can do this from the Job)
+  - ServiceCatalogAPIServer (cr)
+  - ConfigMap named `openshift-service-catalog-apiserver-operator-config`
+  - Service named `metrics`
+  - ClusterRoleBinding for `openshift-service-catalog-apiserver-operator` ServiceAccount
+  - ServiceAccount `openshift-service-catalog-apiserver-operator`
+  - Deployment named `openshift-service-catalog-apiserver-operator`
+  - ClusterOperator named `service-catalog-apiserver`
+- cluster-svcat-controller-manager-operator
+  - Role named `prometheus-k8s in `openshift-service-catalog-controller-manager-operator`
+  - RoleBinding named `prometheus-k8s in `openshift-service-catalog-controller-manager-operator`
+  - ServiceMonitor in `openshift-service-catalog-controller-manager-operator`
+  - Namespace `openshift-service-catalog-controller-manager-operator` (not sure
+    if we can do this from the Job)
+  - ServiceCatalogControllerManager (cr)
+  - ConfigMap named `openshift-service-catalog-controller-manager-operator-config`
+  - Service named `metrics`
+  - ClusterRoleBinding for `openshift-service-catalog-controller-manager-operator` ServiceAccount
+  - ServiceAccount `openshift-service-catalog-controller-manager-operator`
+  - Deployment named `openshift-service-catalog-controller-manager-operator`
+  - ClusterOperator named `service-catalog-controller-manager`
+
+The Job will NOT touch the Service Catalog resources. These resources will
+remain if the admin marked the Service Catalog CRs as `Unmanaged` meaning they
+will self manage the Service Catalog relieving the operators of their duty.
+
+The Service Catalog operators would remain in the release payload but the
```

Comment:
The service catalog operators would be removed right, we would keep one of the images, but change it to the job described above right?

Reply:
@shawn-hurley the service-catalog operator will be removed in the sense that their images and code will be gone. The job will be its own image but is being deployed by the service-catalog operators. My plan was to put a job in BOTH of the service catalog operators, one in the apiserver-operator and the other in the controller-manager-operator. Each job will focus on their items their respective operators are in charge of.

Reply:
So we will be creating two new images, do we need new repos? or are we just going to create new images out of the operator's repos?

We have to get the images apart of the payload, does this require any setup? It may be as simple as label on the image, but IDK.

How do we delete the namespace for the operators, if the jobs are running in those namespaces?

Reply:
@shawn-hurley right now I was going to reuse the existing operator repos because of a conversation we had a long time ago with @wking about adding a job to the manifest directory. But it does feel dirty to use these 2 existing repos for such a thing. And yes the namespace would be a problem.

If we create one new repo for the jobs then they would have their own namespace and do the removal of the svcat stuff. Then we COULD remove the svcat operators from the release altogether.